ScoutSuite/providers/azure/services/network.py

```diff
@@ -0,0 +1,149 @@
+                                                                    sr.source_address_prefixes)
+            security_rule_dict['source_address_prefixes'] = source_address_prefixes
+
+            source_port_ranges = self._merge_prefixes_or_ports(sr.source_port_range, sr.source_port_ranges)
+            security_rule_dict['source_port_ranges'] = source_port_ranges
+            security_rule_dict['source_ports'] = self._parse_ports(source_port_ranges)
+
+            destination_address_prefixes = self._merge_prefixes_or_ports(sr.destination_address_prefix,
+                                                                         sr.destination_address_prefixes)
+            security_rule_dict['destination_address_prefixes'] = destination_address_prefixes
+
+            destination_port_ranges = self._merge_prefixes_or_ports(sr.destination_port_range,
+                                                                    sr.destination_port_ranges)
+            security_rule_dict['destination_port_ranges'] = destination_port_ranges
+            security_rule_dict['destination_ports'] = self._parse_ports(destination_port_ranges)
+
+            security_rule_dict['etag'] = sr.etag
+
+            security_rules[security_rule_dict['id']] = security_rule_dict
+
+        return security_rules
+
+    @staticmethod
+    def _parse_ports(port_ranges):
+        ports = set()
+        for pr in port_ranges:
+            if pr == "*":
+                for p in range(0, 65535 + 1):
+                    ports.add(p)
+                break
+            elif "-" in pr:
+                lower, upper = pr.split("-")
+                for p in range(int(lower), int(upper) + 1):
+                    ports.add(p)
+            else:
+                ports.add(int(pr))
+        ports = list(ports)
+        ports.sort()
+        return ports
+
+    @staticmethod
+    def _parse_exposed_ports(network_security_group):
+        exposed_ports = set()
+
+        # Sort by priority.
+        rules = network_security_group.default_security_rules + network_security_group.security_rules
+        rules.sort(key=lambda x: x.priority, reverse=True)
+
+        for sr in rules:
+            if sr.direction == "Inbound" and (sr.source_address_prefix == "*"
+                                              or sr.source_address_prefix == "Internet"):
+                port_ranges = NetworkConfig._merge_prefixes_or_ports(sr.destination_port_range,
+                                                                     sr.destination_port_ranges)
+                ports = NetworkConfig._parse_ports(port_ranges)
+                if sr.access == "Allow":
+                    for p in ports:
+                        exposed_ports.add(p)
+                else:
+                    for p in ports:
+                        exposed_ports.discard(p)
+        exposed_ports = list(exposed_ports)
+        exposed_ports.sort()
+        return exposed_ports
+
+    @staticmethod
```

Comment:
To be consistent with the rest of the codebase, I'd remove the `@staticmethod`. I agree it makes sense to make those methods static, but `@staticmethod` is used literally once in the whole codebase, so it kind of clashes IMHO.